packages/mongodb-dialects/java-driver/src/main/kotlin/com/mongodb/jbplugin/dialects/javadriver/glossary/JavaDriverDialectParser.kt

```diff
@@ -715,26 +716,20 @@ object JavaDriverDialectParser : DialectParser<PsiElement> {
         return typeOfFirstArg != null && typeOfFirstArg.equalsToText(SESSION_FQN)
     }
 
-    private fun parseComputedExpression(element: PsiElement, createsNewField: Boolean = true): HasValueReference<PsiElement> {
+    private fun parseComputedExpression(element: PsiElement): HasValueReference<PsiElement> {
         val (constant, value) = element.tryToResolveAsConstant()
         return HasValueReference(
             when {
                 constant && value is String -> HasValueReference.Computed(
                     element,
                     type = ComputedBsonType(
-                        BsonAny,
+                        BsonString,
```

Comment:
This is the same suggestion as in the original PR

Reply:
Ah yes, I remember answering here. The thing is that while how we define the expression is a string, the type of the value is not a string right? For example "$year" is not a string, is an int32, so using a string here is not accurate. I feel that until we can infer the type of the expression, we should just assume any. WDYT?

Reply:
keeping BsonAny here because the actual type is not relevant for now, will revert this particular change